tss-esapi/tests/integration_tests/structures_tests/tagged_tests/tagged_ecc_scheme_tests.rs

```diff
@@ -0,0 +1,258 @@
+    (EccSchemeAlgorithm::$scheme_alg_item:ident, HashingAlgorithm::$hash_alg:ident) => {
+        assert_algorithm!(
+            EccSchemeAlgorithm::$scheme_alg_item,
+            EccScheme::$scheme_alg_item(HashScheme::new(HashingAlgorithm::$hash_alg))
+        );
+    };
+    (EccSchemeAlgorithm::$scheme_alg_item:ident) => {
+        assert_algorithm!(
+            EccSchemeAlgorithm::$scheme_alg_item,
+            EccScheme::$scheme_alg_item
+        );
+    };
+    (EccSchemeAlgorithm::$scheme_alg_item:ident, $scheme_expr:expr) => {
+        let actual = $scheme_expr.algorithm();
+        let expected = EccSchemeAlgorithm::$scheme_alg_item;
+        assert_eq!(expected, actual);
+    };
+}
+
+#[test]
+fn test_algorithm_method() {
+    assert_algorithm!(EccSchemeAlgorithm::EcDsa, HashingAlgorithm::Sha256);
+    assert_algorithm!(EccSchemeAlgorithm::EcDh, HashingAlgorithm::Sha256);
+    assert_algorithm!(EccSchemeAlgorithm::EcDaa, HashingAlgorithm::Sha256, 1u16);
+    assert_algorithm!(EccSchemeAlgorithm::Sm2, HashingAlgorithm::Sha256);
+    assert_algorithm!(EccSchemeAlgorithm::EcSchnorr, HashingAlgorithm::Sha256);
+    assert_algorithm!(EccSchemeAlgorithm::EcMqv, HashingAlgorithm::Sha256);
+    assert_algorithm!(EccSchemeAlgorithm::Null);
+}
+
+macro_rules! test_conversions {
+    (EccScheme::$scheme_item:ident, HashingAlgorithm::$hash_item:ident, $count_expr:expr, $details:ident) => {
+        test_conversions!(
+            EccScheme::$scheme_item(EcDaaScheme::new(HashingAlgorithm::$hash_item, $count_expr)),
+            TPMT_ECC_SCHEME {
+                scheme: EccSchemeAlgorithm::$scheme_item.into(),
+                details: TPMU_ASYM_SCHEME {
+                    $details: EcDaaScheme::new(HashingAlgorithm::$hash_item, $count_expr).into(),
+                }
+            }
+        );
+    };
+    (EccScheme::$scheme_item:ident, HashingAlgorithm::$hash_item:ident, $details:ident) => {
+        test_conversions!(
+            EccScheme::$scheme_item(HashScheme::new(HashingAlgorithm::$hash_item)),
+            TPMT_ECC_SCHEME {
+                scheme: EccSchemeAlgorithm::$scheme_item.into(),
+                details: TPMU_ASYM_SCHEME {
+                    $details: HashScheme::new(HashingAlgorithm::$hash_item).into(),
+                }
+            }
+        );
+    };
+    (EccScheme::$scheme_item:ident) => {
+        test_conversions!(EccScheme::$scheme_item, TPMT_ECC_SCHEME {
+            scheme: EccSchemeAlgorithm::$scheme_item.into(),
+            details: Default::default(),
+        });
+    };
+    ($native:expr, $tss:expr) => {
+        let expected_native = $native;
+        let expected_tss = $tss;
+
+        let actual_native = EccScheme::try_from(expected_tss)
+            .unwrap_or_else(|_| {
+                panic!(
+                    "It should be possible to convert a `TPMT_ECC_SCHEME` with valid values into a `{}`.",
+                    std::any::type_name::<EccScheme>(),
+                );
+            });
```

Comment:
Well ... sort of. If you try to use expect with ```format!``` you will get a lint error saying to use ```.unwrap_or_else(|_| panic!(...))``` instead because format is eagerly evaluated while the other one is only evaluated if it actually happens.